### PR TITLE
Fix js currency

### PR DESCRIPTION
**Pull request by ltrls**
#### What? Why?

This PR fixes both problems raised in #1623. The issue was that serializers sent amounts of money as localized strings (`to_money.to_s`) so if the locale used a comma as decimal separator, parsing them in JS to perform operations (in the `/account` page) would fail leading either to a wrong amount (parsing stopped at the comma stripping decimals) or a NaN (not a number) error.
Here I made serializers send the amounts as unlocalized strings, and improved the `localizeCurrency` filter to use i18n-js `toCurrency` method, which as a nice "side effect" also localizes the decimal separator and spacer character.
I18n-js's `toCurrency` method uses translation values in `number.currency.format`, which are set in the rails-i18n gem.

#### What should we test?

The localizeCurrency filter is used in many places but the most notable page is the page at `/account` (see images in related issue) which now presents no error when using a locale with comma as decimal separator. In other places, amounts should also be nicely localized.

### CODE REVIEW

File: app/serializers/api/order_serializer.rb [hunk 1]
```diff
@@ -23,10 +23,6 @@ def changes_allowed_until
       I18n.l(object.order_cycle.andand.orders_close_at, format: "%b %d, %Y %H:%M")
     end
 
-    def total
-      object.total.to_money.to_s
-    end
-
```
Comment (sauloperez) [inline app/serializers/api/order_serializer.rb]: we don't use it anymore?

Comment (ltrls) [reply app/serializers/api/order_serializer.rb]: Yes we do but it would still appear as it's declared in the [`attributes` method](https://github.com/ltrls/openfoodnetwork/blob/c43e40a4c460fb5f7ae352f44a5d5fbc3aec8238/app/serializers/api/order_serializer.rb#L3). What we want is that it's like `def total; object.total; end` and there's no need to have this as it's like not writing the method (customizing the field) at all.

**Comment by sstead** [comment]
**Testing Notes**
- The /Account/Transactions shop 'balance due' numbers are correct, and the tally is calculating correctly.

Some currency display errors we saw...
image

**Comment by myriamboure** [comment]
Are the currency display issues connected to that PR ? Any idea @ltrls ? It seems it is being merged so I suspect it is not related, or should we open another issue ? @sstead this is not clear to me :-o

**Comment by daniellemoorhead** [comment]
@oeoeaio see @myriamboure's comment above. Were the currency display errors fixed by you before you merged it, or where they put into another issue to be done separately?

**Comment by oeoeaio** [comment]
Sorry, I thought I responded to this on Friday. Currency display errors were not fixed, but I've just realised that the reason was probably our user error. The currency components affected by this PR were displaying correctly, but the components being rendered by Rails were not displaying correctly. I now realise that this was probably because we didn't change the configuration of `currency_symbol_position` in the backend. I didn't really think about it (since we don't use it) and just assumed that the locale would change this automatically.

So, in short: @myriamboure, can you confirm that this looks ok for you in France the next time you deploy? There problem is probably just a fairly non-intuitive configuration process, rather than an issue with the logic.

**Comment by myriamboure** [comment]
@oeoeaio I can confirm we just deployed the master version and it's all good on the currency display, the € symbol is after the price on all locations, shopfront, basket, payment, order management, etc. :+1:

**Comment by oeoeaio** [comment]
Awesome, thanks!
